### PR TITLE
feat: Allow passing parameters to perform

### PR DESCRIPTION
**Pull request by lukas-valenta**
## Description
Allows overriding integration parameters in perform call

## Motivation and Context
This allows us to configure specific calls at runtime, like per-user configuration

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have updated the documentation accordingly. If you are changing **code related to user secrets** you need to really make sure that [security documentation](https://github.com/superfaceai/one-sdk-js/blob/main/SECURITY.md) is correct.
- [x] I have read the [CONTRIBUTING](https://github.com/superfaceai/one-sdk-js/blob/main/CONTRIBUTING.md) document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/client/profile-provider.ts [hunk 1]
```diff
@@ -181,6 +185,24 @@ export class BoundProfileProvider {
 
     return ok(result.value);
   }
+
+  private mergeParameters(
+    parameters?: Record<string, string>,
+    providerParameters?: Record<string, string>
+  ): Record<string, string> {
+    if (parameters === undefined) {
+      return providerParameters ?? {};
```
Comment (martinalbert) [inline src/client/profile-provider.ts:194]: Why not possibly return undefined? Map interpreter takes parameters as optional parameter

**Review by martinalbert — APPROVED** [review]

File: src/client/profile-provider.ts (hunk 1 above)
Comment (lukas-valenta) [reply src/client/profile-provider.ts:194]: you're right. to be honest i didn't write this code, copilot did :grin:

Comment (martinalbert) [reply src/client/profile-provider.ts]: oh nice 😄 could it write it correctly if you add undefined as return type too?

Comment (lukas-valenta) [reply src/client/profile-provider.ts]: hard to say, it wrote the return type itself :D

**Pull request merged**